crates/cache-key/src/canonical_url.rs

```diff
@@ -17,6 +17,23 @@ use crate::cache_key::{CacheKey, CacheKeyHasher};
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
 pub struct CanonicalUrl(Url);
 
+trait RemoveCredentials {
+    fn without_credentials(&self) -> Url;
+}
+
+impl RemoveCredentials for Url {
+    fn without_credentials(&self) -> Url {
+        let mut without_creds = self.clone();
+        if without_creds.password().is_some() {
+            let _ = without_creds.set_password(None);
+        }
+        if !without_creds.username().is_empty() {
+            let _ = without_creds.set_username("");
+        }
```

Comment:
Is there any reason to do these checks? Should we just perform the `set` calls?

Reply:
I think we could plausibly make this method return `Cow<'_, Url>` to avoid a clone in the event that there are no credentials. (It would need to continue taking `&self`.)